config.py

```diff
@@ -1,31 +1,48 @@
 """Contains all the configuration of the Flask App"""
 
-import os
-from dotenv import load_dotenv, find_dotenv
-
-load_dotenv(find_dotenv())
-
-MONGO_URI = os.environ.get("MONGO_URI")
-MONGO_TEST_URI = os.environ.get("MONGO_TEST_URI")
-CELERY_BROKER_URL = os.environ.get("CELERY_BROKER_URL")
-CELERY_RESULT_BACKEND = os.environ.get("CELERY_RESULT_BACKEND")
-
-
-CONFIG = {
-    "test": {
-        "MONGODB_SETTINGS": {"host": MONGO_TEST_URI},
-        "CELERY": {
-            "broker_url": CELERY_BROKER_URL,
-            "result_backend": CELERY_BROKER_URL,
-        },
-        # This mode ensures exceptions are propagated rather than handled by the app's error handlers.
-        "TESTING": True,
-    },
-    "prod": {
-        "MONGODB_SETTINGS": {"host": MONGO_URI},
-        "CELERY": {
-            "broker_url": CELERY_BROKER_URL,
-            "result_backend": CELERY_BROKER_URL,
-        },
-    },
-}
+from os import environ
+from pydantic import BaseSettings, Field
+
+
+class DatabaseConfig(BaseSettings):
+    """Database Configurations"""
+
+    HOST: str = Field(env="MONGODB_HOST")
+    PORT: int = Field(27017, env="MONGODB_PORT")
+    DB: str = Field(env="MONGODB_DATABASE")
+    USERNAME: str = Field(env="MONGODB_USERNAME")
+    PASSWORD: str = Field(env="MONGODB_PASSWORD")
+
+    class Config:
+        """Meta Config"""
+
+        env_file = ".env"
+
+
+class CeleryConfig(BaseSettings):
+    """Celery Configurations"""
+
+    BROKER_URL: str = Field(env="CELERY_BROKER_URL")
+    RESULT_BACKEND: str = Field(env="CELERY_RESULT_BACKEND")
+
+    class Config:
+        """Meta Config"""
+
+        env_file = ".env"
+
+
+class BaseConfig(BaseSettings):
+    """Global Configurations"""
+
+    MONGODB_SETTINGS: list = [DatabaseConfig().dict()]
+    CELERY: dict = CeleryConfig().dict()
+
+
+class TestConfig(BaseConfig):
+    """Test Configurations"""
+
+    TESTING: bool = True
```

Comment:
Check if we can use dict as Field and load .env fields inside that. For example,
```python
class DatabaseConfig(BaseSettings):
    MONGODB_SETTINGS: list = [{
        HOST: Field(env="MONGODB_HOST"),
        . . .
    }]
```

Reply:
Actually this is not working, Last time I used os.environ.get() instead of Field. That's why it was working.

Reply:
We are anyways using ENV variables already set by bashrc/docker, then use `os.environ.get()`. Do you think `.env` will be any better than using already set ENV variables?

Reply:
Ok. No, I think bashrc will be fine.

Reply:
Use environ.get() then!